### ISSUE TITLE
Issue with stroke behavior (gif included for review)

### ISSUE THREAD

**Issue by akathewb** [processing/processing#5109]
The issue seems that the strokes loose there references (see gif below)

The github location for this program can be found here: https://github.com/joezcool02/OurCalculator
(This is a collaborative project being run by the folks over at Redit/Processing for leaning how to collaborate on github and git) 

Others have run this same program on there computer without error, so it maybe a computer specific thing to do with the graphics card or render engine. Just want to see if others have seen this before, I searched other forums but came up with no reports.

![weirdstrokeissue](https://cloud.githubusercontent.com/assets/23046037/26759159/2396fc62-48ab-11e7-8306-515a13f891b4.gif)

**Comment by benfry** [comment]
Very strange… Is everyone using the same version of Processing?

**Comment by jeremydouglass** [comment]
This reminds me of a strange time-delayed problem with rect distorting that a user was having on the forums recently. Might be related...?

https://forum.processing.org/two/discussion/22222/solved-rect-stroke-bug/p1

Couldn't reproduce so my best guess was that it was a hardware / driver error. They reported "solving" it by switching renderers....

**Comment by MasterYushi** [comment]
I am running the latest version of processing and couldn't reproduce this.

**Comment by akathewb** [comment]
@benfry
 I'm currently running 3.3.3 64-bit Windows.

I tried some other builds as well below are the results:
2.2.1 64-bit Windows,  No Issue with the strokes, all seems OK!
3.1.1 64-bit Windows, Yes the stokes get messed up
3.2.1 64-bit Windows, Yes the strokes get messed up
3.3    64-bit Windows, Yes the strokes get messed up

Seems like something about rendering in Processing 2 vs Processing 3 might be a little different??

@jeremydouglass 
Below are the results for changing the render mode
FX2D: This seems to work, with no issues
P2D & P3D: This seems to work, but the text letters get pixelated
![image](https://cloud.githubusercontent.com/assets/23046037/26764959/6efc311a-4926-11e7-9358-c4a8de227627.png)


Also, What render mode is used when nothing is specified?

Thanks All!

**Comment by benfry** [comment]
There are (literally) thousands of changes between Processing 2.x and 3.x, but my best guess is that the one that's likely causing this issue is the different version of Java. You could try copying the `java` folder from the 3.x version to 2.x and see if that exhibits the same behavior. (Though of course it's also possible that we broke something along the way… but because it works on other people's machines, that usually means it's something happening outside Processing itself.)

The renderer used by default is the Java2D renderer.

**Comment by akathewb** [comment]
@benfry 
I tried copying the Java folder over from 3.3.3 to 2.2.1 but got the same error described here: https://github.com/processing/processing/issues/2260

I agree with your assessment, it's likely outside processing.

Also, I really appreciate the feedback and thanks developing and supporting such a great program, FYI, I recently joined the processing foundation :)

**Comment by benfry** [comment]
Yeah, sorry it continues to give you trouble. The only other spot where I've seen this is if you're trying to draw at the same time from another thread, but I don't see that happening in your code, so I'm a little baffled. 

If you (or anyone else) figures it out, please let us know.

That's awesome that you've joined as a Member. It's enormously appreciated! Good luck with your collaboration project, it's great to see efforts like yours.
